splipy/SplineObject.py

```diff
@@ -1147,6 +1147,49 @@ def split(self, knots, direction=0):
 
         return results
 
+    def make_periodic(self, continuity=None, direction=0):
+        """Make the spline object periodic in a given parametric direction.
+
+        :param continuity: The continuity along the boundary (default max).
+        :param direction: The direction to ensure continuity in.
+        """
+
+        direction = check_direction(direction, self.pardim)
+        basis = self.bases[direction]
+        if continuity is None:
+            continuity = basis.order - 2
+        if not 0 <= continuity <= basis.order - 2:
+            raise ValueError('Illegal continuity for basis of order {}: {}'.format(
+                continuity, order
+            ))
+        if basis.periodic >= 0:
+            raise ValueError('Basis is already periodic')
+
+        basis = basis.make_periodic(continuity)
+
+        # Merge control points
```

Comment:
Initial thoughts is to just treat the weights as just another controlpoint, i.e. use (self.pardim+self.rational) instead of self.pardim. Can't tell the immediate consequences. Might work. If not, then at least throw warning that this is not supported for rational input.